### ISSUE TITLE
Cannot activate/approve device on PTS server anymore

### ISSUE THREAD

**Issue by github-chrome** [phoronix-test-suite/phoronix-test-suite#388]
Some issues happened with two devices and a PTS server host in my experiment on PTS. (sha1: 7e5c23ea8c7c060884561a0f395f64199659dcb9)

I cannot remember the exact procedure to reproduce the issue so just seek some help in case someone knows any workarounds.

I killed connection from device side when a test failed to finish. After that I cannot bring back the device into idle status on both sides.

I have tried to toggle "enable/disable" and maintenance mode of device at server side. But the status is not synced-up between device and the server. Now the device shows "waiting approval", but the server doesn't show a device is waiting for the approval.

I also noticed there are several PHP zombie  processes left after this mess on the server machine.

Rebooting server machine and relaunching PTS doesn't help. The device to approve is not shown on the server. But there is no way to purge or delete these dead devices instances in order to re-activate them.

I could bring more accurate steps later, but I have observed such issues in several iterations of PTS client - server setup these days.

**Comment by michaellarabel** [comment]
Do you have the systemd process enabled on the clients? Otherwise not aware of any known issues in the Phoromatic client/server for what you describe short of not running the processes on the clients.

**Comment by github-chrome** [comment]
I have one device running on a debian 9 that have several processes of systemd. Another "device" is actually the host machine that also runs the PTS server (ubuntu 18). I am not sure if PTS requres any specific systemd services.

For that pseudo client running on the host, it's still pending on approval. I have tried to enable it in server's web interface. The server web page shows it is active now but still running the last ticket. The connection is not established actually. My impression is the server updates the mode or status immediately without really waiting for device'a ack. (Otherwise, the device would know this and stop waiting the approval). I think it will be nice to have an option to purge inactive devices from server and even have a parameter at device side to clean up.

I will troubleshoot more to get a clear sequence to reproduce the issue.

Thank lot

**Comment by michaellarabel** [comment]
With PTS 9.2.1 is now a fix that there's a chance it could be related to the problem you're experience, can you confirm pulling latest Git is all good? Thanks.
